### PR TITLE
do not require boost if pyosys is not used

### PR DESCRIPTION
**Pull request by mmicko**
Just a small update for macOS builds

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -682,6 +689,7 @@ endif
 
 test: $(TARGETS) $(EXTRA_TARGETS)
 	+cd tests/simple && bash run-test.sh $(SEEDOPT)
+	+cd tests/simple_abc9 && bash run-test.sh $(SEEDOPT)
```
Comment (eddiehung) [inline Makefile]: This looks like an unrelated change?

Comment (mmicko) [reply Makefile]: uf, my bad, let me check, guess I did not update something well
